soda_machine.rb

```diff
@@ -7,12 +7,19 @@ def initialize(args = {})
   end
 
   def current_inventory_count
+    @sodas.length
   end
 
   def find_soda(soda_brand)
+    @sodas.select { |soda| soda.brand == soda_brand } [0]
   end
 
   def sell(soda_brand)
+    selected_soda = sodas.select { |soda| soda.brand == soda_brand } [0]
+    unless selected_soda.nil?
+      @cash += selected_soda.price
+    end
+    sodas.delete(selected_soda)
```

Comment:
This should probably go in the unless as well -- it'll be fine if it's not there, but no point trying to delete it if you know it's not there.